Update inheritance-mapping.rst (#8704)
doctrine/doctrine2: docs/en/reference/inheritance-mapping.rst

Otherwise Doctrine *CANNOT* create proxy instances
of this entity and will *ALWAYS* load the entity eagerly.

Also, there is another important performance consideration: It is *NOT POSSIBLE* 

> to query for the base entity avoiding all the LEFT JOINs with the sub-types.

Reviewer: Not sure I understand this sentence considering the linked issue that led to this PR. Do you suggest to mention that it is impossible to query on a CTI without having no LEFT JOINS?

Author: Yes. What I tried to specify is that you can't query for the base entity of the inheritance without avoiding all the left joins with the leafs of the inheritance tree.

Reviewer: ```suggestion
There is also another important performance consideration that it is *NOT POSSIBLE* 
to query for the base entity without any LEFT JOINs to the sub-types.
```

Maybe this is more understandable. What do you think?

Author: Agreed! I will commit that
